src/MycilaESPConnect_CP.cpp

```diff
@@ -89,18 +89,24 @@ void Mycila::ESPConnect::_startCaptivePortal() {
       } else {
         ESPCONNECT_STRING ssid;
         ESPCONNECT_STRING password;
+        ESPCONNECT_STRING bssid;
         if (request->hasParam("ssid", true))
           ssid = request->getParam("ssid", true)->value().c_str();
         if (request->hasParam("password", true))
           password = request->getParam("password", true)->value().c_str();
+        if (request->hasParam("bssid", true))
+          bssid = request->getParam("bssid", true)->value().c_str();
         if (!ssid.length())
           return request->send(400, "application/json", "{\"message\":\"Invalid SSID\"}");
         if (ssid.length() > 32 || password.length() > 64 || (password.length() && password.length() < 8))
           return request->send(400, "application/json", "{\"message\":\"Credentials exceed character limit of 32 & 64 respectively, or password lower than 8 characters.\"}");
-        _config.wifiSSID = ssid;
-        _config.wifiPassword = password;
-        request->send(200, "application/json", "{\"message\":\"Configuration Saved.\"}");
-        _setState(Mycila::ESPConnect::State::PORTAL_COMPLETE);
+        if (_credentialTest.request != nullptr)
+          return request->send(409, "application/json", "{\"message\":\"A connection test is already in progress. Please wait.\"}");
+        // Try WiFi connection test before saving and closing portal
+        const uint32_t testTimeoutSec = 5; // reduced timeout to prevent watchdog issues
```

Comment:
> // reduced timeout to prevent watchdog issues
that's not right. f the TWDT triggers it means that there is a coding issue. Mainly because a long task executes within the context of one that must execute fast.

I think this is just a a matter of re-organizing the code a little.